book_maker/translator/chatgptapi_translator.py

```diff
@@ -388,3 +407,224 @@ def set_model_list(self, model_list):
         model_list = list(set(model_list))
         print(f"Using model list {model_list}")
         self.model_list = cycle(model_list)
+
+    def batch_init(self, book_name):
+        self.book_name = self.sanitize_book_name(book_name)
+
+    def add_to_batch_translate_queue(self, book_index, text):
+        self.batch_text_list.append({"book_index": book_index, "text": text})
+
+    def sanitize_book_name(self, book_name):
+        # Replace any characters that are not alphanumeric, underscore, hyphen, or dot with an underscore
+        sanitized_book_name = re.sub(r"[^\w\-_\.]", "_", book_name)
+        # Remove leading and trailing underscores and dots
+        sanitized_book_name = sanitized_book_name.strip("._")
+        return sanitized_book_name
+
+    def batch_metadata_file_path(self):
+        return f"{os.getcwd()}/batch_files/{self.book_name}_info.json"
+
+    def batch_dir(self):
+        return f"{os.getcwd()}/batch_files/{self.book_name}"
```

Comment:
ditto seems not support windows file name

Reply:
I fixed it
[f1e78eb](https://github.com/yihong0618/bilingual_book_maker/pull/423/commits/f1e78ebcb940b48a2e6d97579990d95e40fa7f11)